[M2-6246] Rename Participant "role" to "tag"

farmerpaul opened the pull request:
<!-- Use this template as a guide to describe your pull request, and adjust as necessary. -->
<!-- Include information that helps your peers review your updates and understand this    -->
<!-- repository's history of changes over time.                                           -->

<!-- Delete any options that are not relevant -->

- [x] Tests for the changes have been added

### 📝 Description

🔗 [Jira Ticket M2-6246](https://mindlogger.atlassian.net/browse/M2-6246)

This PR includes some very minor changes, partly because the desired changes were already mostly done, and because tags are mostly still unimplemented:
- Where referring to a Participant's "tag" (for which BE support was recently added), use the word "Tag" instead of "Role", both in the code and the UI
  - Note: in Figma it's still incorrectly shown as "Role" – that's due to be fixed in [this task](https://mindlogger.atlassian.net/browse/M2-6271)
- Move the "Tag" column to the correct location on the Participants screen

Also, unrelated:
- Fix a TypeScript error introduced in the last merge

### 📸 Screenshots

<!--
If your work here contains visual changes, provide before (optional) and after screenshots, GIFs, or videos.

If not, then delete this section
-->

| Before                      | After                                 |
| -------------------------------------- | ------------------------------------- |
| <img width="1306" alt="Screenshot 2024-04-16 at 5 10 01 PM" src="https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/76d71e09-c9ac-4726-9380-b71a8f529969"> | <img width="1309" alt="Screenshot 2024-04-16 at 5 09 01 PM" src="https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/58c472af-b97b-468b-a6ff-a259b42757c3"> |
| <img width="730" alt="Screenshot 2024-04-16 at 5 09 48 PM" src="https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/b67311c0-8560-41f9-8b5d-36d7e04edd33"> | <img width="730" alt="Screenshot 2024-04-16 at 5 09 24 PM" src="https://github.com/ChildMindInstitute/mindlogger-admin/assets/1282772/f2ffed1d-3405-4939-9456-2ed7757f217c"> |

farmerpaul committed:
M2-6246: fix tsc errors

Assign ParticipantSnippetProps default type argument in props definition
to simplify usage.

farmerpaul committed:
M2-6246: update role/label language with "tag"

Also move the Participants table Tag column to the correct position.

sultanofcardio commented on src/modules/Dashboard/components/ParticipantSnippet/ParticipantSnippet.tsx:
```diff
@@ -1,18 +1,15 @@
 import { ParticipantSnippetProps } from './ParticipantSnippet.types';
 
-export const ParticipantSnippet = <T extends ElementType = BoxTypeMap['defaultComponent']>({
+export const ParticipantSnippet = ({
```
We have to keep the type parameter here, otherwise typescript complains about this props spread

https://github.com/ChildMindInstitute/mindlogger-admin/blob/7460397c388471a5692244087f997ce2094612c2/src/modules/Dashboard/components/TakeNowModal/LabeledDropdown/LabeledUserDropdown.tsx#L83-L89

I thought I'd be able to get around this by specifying the `component` prop, but the typing information doesn't seem to propagate to the `ParticipantSnippetProps`, which maintains its default value

farmerpaul replied:
Yup, I can see why this would fail.

What do you think about just making do with `ParticipantSnippet` always rendering a `div`, and wrapping yours in an `<li>`, just to keep things a little simpler? Otherwise I think we'll have to propagate that awkward generic syntax in several contexts… maybe it's not as bad as I think, but it seemed to be having a bit of a domino effect when I was trying to resolve it earlier.

sultanofcardio replied:
I can live with wrapping it in a `<li>`, but I think the generic syntax is only necessary if you're trying to spread props that you don't control. Like in this instance, the Autocomplete is expecting a list item element. Do you foresee this applying anywhere else?

farmerpaul replied:
I solved it by doing a minor refactor (extracted model props from the component props), so I was able to preserve the generic syntax and it shouldn't affect your code. :boom:

farmerpaul committed:
M2-6246: small refactor of ParticipantSnippetProps

Separated model props from UI props so that the generic functionality
of the component can still be used.

farmerpaul committed:
M2-6246: streamline UpgradeAccountPopup import
